cmd/gin-repod/repo.go

```diff
@@ -54,10 +54,17 @@ func (s *Server) createRepo(w http.ResponseWriter, r *http.Request) {
 
 	rid := store.RepoId{Owner: owner, Name: creat.Name}
 
-	_, ok := s.checkAccess(w, r, rid, store.NoAccess)
+	user, ok := s.checkAccess(w, r, rid, store.NoAccess)
 	if !ok {
 		return
 	}
+	// make sure routes user and token user are identical
+	if owner != user.Uid {
```

Comment:
I believe that `ok` can be `true` and `user` could be `nil` at the same time, which would lead to a null pointer dereference at `user.Uid`.